ibis/backends/base/sqlglot/__init__.py

```diff
@@ -1,97 +1,246 @@
-        self.aggfunc = aggfunc
-
-    def __getattr__(self, name: str) -> partial:
-        return partial(self.aggfunc, name)
-
-    def __getitem__(self, key: str) -> partial:
-        return getattr(self, key)
-
-
-def _func(name: str, *args: Any, **kwargs: Any):
-    return sg.func(name, *map(sg.exp.convert, args), **kwargs)
-
-
-class FuncGen:
-    __slots__ = ()
-
-    def __getattr__(self, name: str) -> partial:
-        return partial(_func, name)
-
-    def __getitem__(self, key: str) -> partial:
-        return getattr(self, key)
-
-    def array(self, *args):
-        return sg.exp.Array.from_arg_list(list(map(sg.exp.convert, args)))
-
-    def tuple(self, *args):
-        return sg.func("tuple", *map(sg.exp.convert, args))
-
-    def exists(self, query):
-        return sg.exp.Exists(this=query)
-
-    def concat(self, *args):
-        return sg.exp.Concat(expressions=list(map(sg.exp.convert, args)))
+import ibis
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+from ibis.backends.base import BaseBackend
+from ibis.backends.base.sqlglot.compiler import STAR
 
-    def map(self, keys, values):
-        return sg.exp.Map(keys=keys, values=values)
-
-
-class ColGen:
-    __slots__ = ()
-
-    def __getattr__(self, name: str) -> sg.exp.Column:
-        return sg.column(name)
-
-    def __getitem__(self, key: str) -> sg.exp.Column:
-        return sg.column(key)
-
-
-def paren(expr):
-    """Wrap a sqlglot expression in parentheses."""
-    return sg.exp.Paren(this=expr)
-
-
-def parenthesize(op, arg):
-    import ibis.expr.operations as ops
-
-    if isinstance(op, (ops.Binary, ops.Unary)):
-        return paren(arg)
-    # function calls don't need parens
-    return arg
-
-
-def interval(value, *, unit):
-    return sg.exp.Interval(this=sg.exp.convert(value), unit=sg.exp.var(unit))
-
-
-C = ColGen()
-F = FuncGen()
-NULL = sg.exp.NULL
-FALSE = sg.exp.FALSE
-TRUE = sg.exp.TRUE
-STAR = sg.exp.Star()
-
-
-def make_cast(
-    converter: SqlglotType,
-) -> Callable[[sg.exp.Expression, dt.DataType], sg.exp.Cast]:
-    def cast(arg: sg.exp.Expression, to: dt.DataType) -> sg.exp.Cast:
-        return sg.cast(arg, to=converter.from_ibis(to))
+if TYPE_CHECKING:
+    from collections.abc import Iterator
 
-    return cast
+    import ibis.expr.datatypes as dt
+    import ibis.expr.types as ir
+    from ibis.backends.base.sqlglot.compiler import SQLGlotCompiler
+    from ibis.common.typing import SupportsSchema
+
+
+class SQLGlotBackend(BaseBackend):
+    compiler: ClassVar[SQLGlotCompiler]
+    name: ClassVar[str]
+
+    @classmethod
+    def has_operation(cls, operation: type[ops.Value]) -> bool:
+        # singledispatchmethod overrides `__get__` so we can't directly access
+        # the dispatcher
+        dispatcher = cls.compiler.visit_node.register.__self__.dispatcher
+        return dispatcher.dispatch(operation) is not dispatcher.dispatch(object)
+
+    def _transform(
```

Comment:
What does this method do/responsible for?

Reply:
It's implemented in the DuckDB backend to add the necessary `st_aswkb` calls to any geometry columns in the final select expression. This **must** be done as the last step, and cannot be done as an ibis rewrite, because we only want to rewrite the final projection that the user executes, not every projection with geometry columns.

Reply:
Could we rather override `_to_sqlglot()` in the duckdb compiler?